### ISSUE TITLE
formatter: CSS declaration followed by a block fails to parse, but Prettier formats it

### ISSUE THREAD

**Issue by miloag** [oxc-project/oxc#26158]
### What happens

Oxfmt rejects a CSS declaration that is followed by a block. Prettier formats the same input, so this is a Prettier-compatibility gap rather than invalid input.

```css
@font {
  sans: "Sans" {
    weight: 100 900;
  }
}
```

```
  x Syntax error: component value is expected
   ,-[src/index.css:2:16]
 1 | @font {
 2 |   sans: "Sans" {
   :                ^
 3 |     weight: 100 900;
   `----
```

Prettier 3.8 formats it unchanged. PostCSS reads `sans: "Sans"` as a rule prelude, so every PostCSS-based tool accepts the shape.

The same shape inside `@animate` fails the same way:

```css
@animate {
  wiggle: wiggle 1s ease-in-out infinite {
    from { transform: rotate(-3deg); }
  }
}
```

### Why it matters

A parse error fails the whole file, and `vp check` stops there — before the lint and type-check steps run. Ignore comments do not help, because the parse fails before any comment is read, so the only escape is `fmt.ignorePatterns`, which drops the file from formatting altogether.

CSS dialects built on PostCSS use this shape for nested configuration blocks. The example above is [Rainbow Index](https://github.com/rainbowindex/rainbowindex)'s `@font` directive, but any postcss-nested-style dialect can produce it.

### What I am not asking for

Three other forms in the same dialect are rejected by Prettier too, so they look out of scope: `!name;` removals, bare keyword statements (`parabolic;`), and a `{` group inside an at-rule prelude (`@apply hover:{px-2 py-1};`).

### Versions

- oxfmt 0.64.0 (through vite-plus 0.3.0)
- prettier 3.8 for the comparison
- macOS, Node 24
